### PR TITLE
Feature/671 delegation

### PR DESCRIPTION
**Pull request by skuhlmann**
## GitHub Issue

fixes https://github.com/HausDAO/daohaus-monorepo/issues/671

## Changes

- member profile menu  used in list and detail page
- dialog form for delegate
- all the delegate form and tx legos
- new delegate field

## Packages Added

n/a

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [ ] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/core-app/src/components/MemberProfileMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  Theme,
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  Button,
+  DropdownLink,
+  DropdownText,
+} from '@daohaus/ui';
+import { useMemo } from 'react';
+import { BiDotsVerticalRounded } from 'react-icons/bi';
```
Comment (0xRowdy) [inline apps/core-app/src/components/MemberProfileMenu.tsx:16]: Is there a Remix Icon for something similar? 
Sounds like that is the icon pack we're trying to use exclusively

File: apps/core-app/src/components/customFields/delegateInput.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      <WrappedInput {...props} />
+      <ParMd>Voting token to delegate</ParMd>
+      <DataMd>{`${dao?.shareTokenName} (${formatValueTo({
+        value: fromWei(connectedMembership?.shares || '0'),
+        decimals: 2,
+        format: 'number',
+      })} - ${votingPowerPercentage(
+        dao?.totalShares || '0',
+        connectedMembership?.shares || '0'
+      )}% voting power) `}</DataMd>
```
Comment (0xRowdy) [inline apps/core-app/src/components/customFields/delegateInput.tsx:57]: This is a little hard to follow.
What about formatting above and assigning to a variable to use in the markup?

**Review by 0xRowdy — APPROVED** [review]

File: apps/core-app/src/components/Profile.tsx [hunk 3]
```diff
@@ -134,6 +145,7 @@ export const Profile = ({ profile, membership }: ProfileProps) => {
             )}
           </Container>
         </ProfileMetadataContainer>
+        <MemberProfileMenu memberAddress={membership?.memberAddress} />
```
Comment (0xRowdy) [inline apps/core-app/src/components/Profile.tsx:148]: 'string | undefined' error. Remove ? if it will always have a value

**Review by 0xRowdy — APPROVED** [review]

File: apps/core-app/src/components/MemberProfileMenu.tsx (hunk 1 above)
Comment (skuhlmann) [reply apps/core-app/src/components/MemberProfileMenu.tsx]: found it
